app/uk/gov/hmrc/ctcguaranteebalanceeisstub/models/TestScenarios.scala

```diff
@@ -0,0 +1,88 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.ctcguaranteebalanceeisstub.models
+
+import play.api.libs.json.Json
+import play.api.mvc.Result
+import play.api.mvc.Results.*
+import uk.gov.hmrc.ctcguaranteebalanceeisstub.models.AccessCode
+import uk.gov.hmrc.ctcguaranteebalanceeisstub.models.GuaranteeReferenceNumber
+import uk.gov.hmrc.ctcguaranteebalanceeisstub.models.TestData
+import uk.gov.hmrc.ctcguaranteebalanceeisstub.models.responses.AccessCodeResponse
+import uk.gov.hmrc.ctcguaranteebalanceeisstub.models.responses.Balance
+import uk.gov.hmrc.ctcguaranteebalanceeisstub.models.responses.BalanceResponse
+import uk.gov.hmrc.ctcguaranteebalanceeisstub.models.responses.RequestErrorResponse
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+object TestScenarios {
+
+  private def timestamp: String = LocalDateTime.now().format(DateTimeFormatter.ISO_DATE_TIME)
+  private def path: String      = "/guarantees/..."
+
+  private val guaranteeNotFoundGenerator: GuaranteeReferenceNumber => Result = grn =>
+    NotFound(Json.toJson(RequestErrorResponse(s"Guarantee not found for GRN: ${grn.value}", timestamp, path)))
+
+  private val invalidAccessCodeGenerator: Result =
+    Forbidden(Json.toJson(RequestErrorResponse("Not Valid Access Code for this operation", timestamp, path)))
+
+  private val invalidGuaranteeTypeGenerator: Result =
+    Forbidden(Json.toJson(RequestErrorResponse("Not Valid Guarantee Type for this operation", timestamp, path)))
+
+  private val simpleGuaranteeTestDataList: List[TestData] = List(
+    // GB data
+    TestData(0, "23GB0000010000854", BigDecimal("262626"), "GBP", "AC01"),
+    TestData(1, "23GB0000010000863", BigDecimal("10000"), "GBP", "AC01"),
+    TestData(9, "23GB0000010000872", BigDecimal("90000"), "GBP", "AC01"),
+    // XI data
+    TestData(0, "23XI0000010000655", BigDecimal("30000"), "GBP", "AC01"),
+    TestData(1, "23XI0000010000664", BigDecimal("50000"), "GBP", "AC01")
+  )
+
+  private val invalidBalanceCheckTypes: Set[Int] = Set(0, 2, 4, 9)
+
+  val balanceRequestScenarios: Map[GuaranteeReferenceNumber, Either[Result, BalanceResponse]] =
+    simpleGuaranteeTestDataList.map {
+      data =>
+        val grnModel = GuaranteeReferenceNumber(data.grn)
+        val result: Either[Result, BalanceResponse] =
+          if (invalidBalanceCheckTypes.contains(data.guaranteeType)) {
+            Left(invalidGuaranteeTypeGenerator)
+          } else {
+            Right(BalanceResponse(grnModel, Balance(data.amount.doubleValue), data.currency))
+          }
+        grnModel -> result
+    }.toMap
```

Comment:
This instinctively feels like it could be a bit more performant. Rather than generating the response for every possible test data, can we pass in the grn from the controller and just get the response we're interested in?